engineering/teams/methodology.md

```diff
@@ -0,0 +1,135 @@
+
+# Internal Team methodology
+
+Date: 2018-29-11
+
+* [Intro](#intro)
+* [Definitions](#definitions)
+  * [What does TODO mean?](#what-does-todo-mean)
+  * [What does Done mean?](#what-does-done-mean)
+  * [What does Blocker mean?](#what-does-blocker-mean)
+
+## Intro
+
+As guidance to the Leads, this document exposes the agreement about how to apply Kanban for the internal organization of the team. Every team can base his Kanban on this document or create his version.
+
+Github Projects board will used as a Kanban board.
+
+Teams managing issues across Github Org borders (Applications, Language Analysis)
+will use the “add card”  trick, described in dear-github/dear-github#209 to add
+issues from different org  to the team’s board.
+
+Issues repositories:
+
+* [src-d/issues-general](https://github.com/src-d/issues-general) - should be used more, as place for feedback/requests (i.e website, etc)
+* [src-d/issues-infrastructure](https://github.com/src-d/issues-infrastructure) - issues/requests to Infrastructure team must be created here.
+
+# Definitions
+
+Below, we'll define the meaning of each Kanban column.
+
+This is not mere a formality - it’s important to have same things in mind when
+moving work from one state to another and what is documented below is a
+consensus, that took few hours to build among Engineering leads.
+
+## What does TODO mean?
+
+Something an engineer can immediately take and act up on. If the task is not
+possible to start until another (well-known) task is finished first, this task
+shall not be in TODO yet.
+
+*Rationale* there are 2 goals we want to keep in mind, while working with Kanban
+board:
+
+* as small TODO list on a Kanban as possible and
+* tasks, moving fast from end-to-end of kanban board (short Cycle Time)
+
+This definition and goals also imply that leads should grooming backlog on the
+daily basis.
+
+> Alternative definition (discussed but discarded): all things that are planned to get done ASAP. This would imply that issues can be added to TODO in a bulk at once, even in case when multiple issues of single project have known dependency on one another. Was discarded on the premise that it would go against the 2 goals, listed above.
+
+## What does Done mean?
+
+General mindset: Done always means that a user, both inside and outside src-d,
+is able to use it through well-established interfaces. Detailed definition may
+vary per-team.
+
+Few concrete examples, in case of work done on:
+
+* Golang library - it should be possible to use them though the particular version published on gopkg.in
+* Scala library - it should be possible to use from Maven/sbt build
+* Web site/application - should be accessible through a known (staging/production) environment URL
+* etc
+
+
+## What does Blocker mean?
+
+*General mindset*: blocker means something **unexpected has happened**
```

Comment:
I have seen unconsistent usage of these terms:

* **blockeR** should be something that requires urgent action because it's blocking important work by others.
* **bockeD** should be something that cannot be acted upon because it's blocked by something else.